CMakePresets.json

```diff
@@ -10,24 +10,252 @@
       "name": "base",
       "hidden": true,
       "generator": "Ninja",
-      "binaryDir": "${sourceDir}/build/${presetName}",
+      "binaryDir": "${sourceDir}/build/$env{CCCL_BUILD_INFIX}/${presetName}",
       "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
         "CCCL_ENABLE_LIBCUDACXX": false,
         "CCCL_ENABLE_CUB": false,
         "CCCL_ENABLE_THRUST": false,
         "CCCL_ENABLE_TESTING": false,
-        "CCCL_ENABLE_EXAMPLES": false
+        "CCCL_ENABLE_EXAMPLES": false,
+        "libcudacxx_ENABLE_INSTALL_RULES": true,
+        "CUB_ENABLE_INSTALL_RULES": true,
+        "THRUST_ENABLE_INSTALL_RULES": true
       }
     },
     {
-      "name": "examples",
-      "displayName": "CCCL Examples",
-      "description": "Configure, build, and run the CCCL Examples",
+      "name": "libcudacxx-base",
+      "hidden": true,
       "inherits": "base",
       "cacheVariables": {
-        "CMAKE_CUDA_ARCHITECTURES": "70",
-        "CCCL_ENABLE_EXAMPLES": true
+        "CCCL_ENABLE_LIBCUDACXX": true,
+        "LIBCUDACXX_ENABLE_LIBCUDACXX_TESTS": true
+      }
+    },
+    {
+      "name": "libcudacxx-cpp11",
+      "displayName": "libcu++: C++11 (requires manual usage of lit to build/test)",
+      "inherits": "libcudacxx-base",
+      "cacheVariables": {
+        "CMAKE_CXX_STANDARD": "11",
+        "CMAKE_CUDA_STANDARD": "11"
+
+      }
+    },
+    {
+      "name": "libcudacxx-cpp14",
+      "displayName": "libcu++: C++14 (requires manual usage of lit to build/test)",
+      "inherits": "libcudacxx-base",
+      "cacheVariables": {
+        "CMAKE_CXX_STANDARD": "14",
+        "CMAKE_CUDA_STANDARD": "14"
+
+      }
+    },
+    {
+      "name": "libcudacxx-cpp17",
+      "displayName": "libcu++: C++17 (requires manual usage of lit to build/test)",
+      "inherits": "libcudacxx-base",
+      "cacheVariables": {
+        "CMAKE_CXX_STANDARD": "17",
+        "CMAKE_CUDA_STANDARD": "17"
+
+      }
+    },
+    {
+      "name": "libcudacxx-cpp20",
+      "displayName": "libcu++: C++20 (requires manual usage of lit to build/test)",
+      "inherits": "libcudacxx-base",
+      "cacheVariables": {
+        "CMAKE_CXX_STANDARD": "20",
+        "CMAKE_CUDA_STANDARD": "20"
+      }
+    },
+    {
+      "name": "libcudacxx-nvrtc-base",
+      "hidden": true,
+      "inherits": "libcudacxx-base",
+      "cacheVariables": {
+        "LIBCUDACXX_TEST_WITH_NVRTC": true
+      }
```

Comment:
This adds only a single additional cmake flag. Should we just drop it and add it in `nvrtc_libcudacxx`?

Reply:
I think a good goal is to put as little logic in the scripts as possible, because anything in the scripts needs to be replicated in both Linux and Windows scripts.